Rewrite installation doc for more detailed guide (#958)
oam-dev/kubevela.io: docs/installation/standalone.mdx

---
title: Standalone Installation
---

import Tabs from '@theme/Tabs';
import TabItem from '@theme/TabItem';

## Overview

This documents will show you how to install KubeVela in standalone mode. There are two options:

> 1. [Local machine](#local): Install in a local machine.

Reviewer: ```suggestion
1. [Local machine](#local): Install on a local machine based on Linux, MacOS or Windows operating system.
```

Author: Little tips here: the standard writing of "MacOS" is actually "macOS"

> 2. [Remote Linux server](#remote): Install on a remote Linux server with accessible IP.

Reviewer: ```suggestion
2. [Remote Linux server](#remote): Install on a remote Linux server with accessible IP address.
```

### About VelaD

We'll use [VelaD](https://github.com/kubevela/velad) to install KubeVela in standalone mode. VelaD is a CLI tool that
packages KubeVela with all dependencies for minimal installation and VelaUX.
VelaD enables you to run KubeVela on any of docker compatible environments or Linux system.

- VelaD provides Kubernetes by leveraging [K3s](https://rancher.com/docs/k3s/latest/en/quick-start/) on linux or [k3d](https://k3d.io/) on docker environment.
- KubeVela along with all related images, and `vela` command line are packaged together that enables air-gapped installation of KubeVela and VelaUX.

:::caution

> Installation with standalone mode is suited for development and testing purpose only. It is not recommended to use it in production environment.

Reviewer: ```suggestion
Installation with standalone mode is suited for development and testing purpose best that you don't even aware of the Kubernetes inside. If you want to use it in production, you need to understand and maintain the K3s for Kubernetes management well.
```

:::

## Install in a local machine {#local}

### 1. Prepare environment

> You can install KubeVela on machines with these OS: Linux, macOS, Windows。

Reviewer: ```suggestion
You can install KubeVela on machines with these OS: Linux, macOS, Windows.
```

**Requirements**

> 1. Make sure your machine have `curl` installed.

Reviewer: does windows need it?

Author: yes, AFAIK, powershell have "curl" inside.

Author: > yes, AFAIK, powershell have "curl" inside.

I'm wrong. Powershell have another command for http request.

> 2. If you are using macOS or Windows, you need to install [Docker Desktop](https://www.docker.com/products/docker-desktop) first.

Reviewer: ```suggestion
2. If you are using macOS or Windows, make sure you've already installed [Docker](https://www.docker.com/products/docker-desktop).
```

### 2. Install VelaD and Setup KubeVela

#### Download VelaD

It will download and place the binary in your system `PATH`,

<Tabs className="unique-tabs" groupId="os">
<TabItem label="Linux/macOS" value="unix">

You may be required for root privilege during the installation process if not using root.
If you don't need the automation with root access, you can download from the [release page](https://github.com/kubevela/velad/releases) and uncompress manually.

```shell
curl -fsSl https://static.kubevela.net/script/install-velad.sh | bash
```

Check `velad` installed in `/usr/local/bin/`

</TabItem>
<TabItem label="Windows" value="win">

Run following command in powershell with administrator privilege.

```powershell
powershell -Command "iwr -useb https://static.kubevela.net/script/install-velad.ps1 | iex"
```
Check `velad` installed in `C:\vela\`

:::info

> VelaD helps install Vela CLI, which requires administrator.

Reviewer: ```suggestion
VelaD helps install Vela CLI, which requires administrator privileges.
```

:::

<details>

> <summary>How to start powershell with administrator privilege</summary>

Reviewer: ```suggestion
<summary>Start powershell as administrator</summary>
```

  k8s-objects           	autodetects.core.oam.dev	K8s-objects allow users to specify raw K8s objects in
  properties
  raw                   	autodetects.core.oam.dev	Raw allow users to specify raw K8s object in properties.
  This definition is DEPRECATED, please use 'k8s-objects'
  instead.
  ref-objects           	autodetects.core.oam.dev	Ref-objects allow users to specify ref objects to use.
  Notice that this component type have special handle logic.
  ```

</details>
Export kubeconfig and list all built-in ComponentDefinition:

### 3. Install VelaUX

VelaUX is a dashboard for KubeVela. It is a web application that runs in your cluster. You can access it with your browser.
This is optional if you don't use UI console of KubeVela.

<Tabs className="unique-tabs" groupId="os">
  <TabItem label="Linux/macOS" value="unix">

  ```shell script
  vela addon enable ~/.vela/addons/velaux
  ```

  </TabItem>
  <TabItem label="Windows" value="win">

  ```powershell
  vela addon enable $HOME\.vela\addons\velaux
  ```

  </TabItem>
</Tabs>

<details>
  <summary>expected output</summary>

  ```
  Addon: velaux enabled Successfully.
  ```

</details>

> By default, velaux didn't have any exposed port, you can view it by:

Reviewer: ```suggestion
By default, velaux don't have any exposed port, you can view it by:
```

```shell
vela port-forward addon-velaux -n vela-system 8080:80
```

Choose `> local | velaux | velaux` for visit.

:::note
For more VelaUX options, refer to [VelaUX addon](../reference/addons/velaux) document for other advanced installation arguments.
:::

VelaUX needs authentication. The default username is `admin` and the password is **`VelaUX12345`**.

It requires you to override with a new password for the first login, please make sure to remember the new password.

### 4. Cleanup

This command will clean up KubeVela controllers along with the Kubernetes cluster, refer to [the advanced guide](../platform-engineers/advanced-install#uninstall) for more detailed steps.

```
velad uninstall
```

## Install on a remote Linux server {#remote}

> This documents show you how to install KubeVela on a remote Linux server (e.g. aws EC2 or aliyun ECS) in standalone mode.

Reviewer: Let's add some tips to explain why we don't support other OS for remote server.

Author: done

---
title: Standalone Installation
---

import Tabs from '@theme/Tabs';
import TabItem from '@theme/TabItem';

## Overview

This documents will show you how to install KubeVela in standalone mode. There are two options:

1. [Local machine](#local): Install on a local machine based on Linux, macOS or Windows operating system.
2. [Remote Linux server](#remote): Install on a remote Linux server with accessible IP address.

:::tip

> Install KubeVela on a remote server with non-linux operating system is not fully tested. You can try it but we don't guarantee it will work.

Reviewer: ```suggestion
Install KubeVela on a remote server with non-linux operating system is not fully tested.
```
